thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/anomaly/detection/trigger/utils/DatasetTriggerInfoRepo.java

```diff
@@ -0,0 +1,121 @@
+      t.setDaemon(true);
+      return t;
+    });
+    this.executorService.scheduleAtFixedRate(
+        this::refresh, refreshFreqMin, refreshFreqMin, TimeUnit.MINUTES);
+  }
+
+  public static DatasetTriggerInfoRepo initAndGetInstance(int refreshFreqMin, Collection<String> dataSourceWhitelist) {
+    if (_instance == null) {
+      synchronized (DatasetTriggerInfoRepo.class) {
+        if (_instance == null) {
+          _instance = new DatasetTriggerInfoRepo(refreshFreqMin, dataSourceWhitelist);
+        }
+      }
+    }
+    return _instance;
+  }
+
+  public static DatasetTriggerInfoRepo getInstance() {
+    if (_instance == null)
+      throw new IllegalStateException("DatasetTriggerInfoRepo is not initialized before getInstance is called.");
+    return _instance;
+  }
+
+  public boolean isDatasetActive(String dataset) {
+    return datasetRefreshTimeMap.containsKey(dataset);
+  }
+
+  public long getLastUpdateTimestamp(String dataset) {
+    return (isDatasetActive(dataset)) ? datasetRefreshTimeMap.get(dataset) : 0;
+  }
+
+  public void setLastUpdateTimestamp(String dataset, long timestamp) {
+    datasetRefreshTimeMap.put(dataset, timestamp);
+  }
+
+  public void close() {
+    executorService.shutdown();
+    _instance = null;
+  }
+
+  private void refresh() {
+    Set<String> activeDatasetSet = new HashSet<>();
+    DetectionConfigManager detectionConfigDAO = DAO_REGISTRY.getDetectionConfigManager();
+    MetricConfigManager metricsConfigDAO = DAO_REGISTRY.getMetricConfigDAO();
+    DatasetConfigManager datasetConfigDAO = DAO_REGISTRY.getDatasetConfigDAO();
+    List<DetectionConfigDTO> detectionConfigDTOs = detectionConfigDAO.findAll();
+    LOG.info(String.format("Found %d detection configs", detectionConfigDTOs.size()));
+    for (DetectionConfigDTO detectionConfigDTO : detectionConfigDTOs) {
+      if (!detectionConfigDTO.isActive()) continue; // skip inactive detection
+      List<String> metricUrns = DetectionConfigFormatter
+          .extractMetricUrnsFromProperties(detectionConfigDTO.getProperties());
+      if (metricUrns.size() < 1) {
+        LOG.error("Empty metricUrn for detection {}", detectionConfigDTO.getId());
+        continue;
+      }
+      for (String urn : metricUrns) {
+        MetricEntity me = MetricEntity.fromURN(urn);
+        MetricConfigDTO metricConfig = metricsConfigDAO.findById(me.getId());
+        if (metricConfig == null) {
+          LOG.warn("Found null value for metric: " + me.getId());
+          continue;
+        }
+        DatasetConfigDTO datasetConfig = datasetConfigDAO.findByDataset(metricConfig.getDataset());
+        String dataset = datasetConfig.getDataset();
+        if (!dataSourceWhitelist.contains(datasetConfig.getDataSource())) {
+          LOG.info("Skip detection {} because {} data source is not whitelisted", detectionConfigDTO.getId(), dataset);
+          continue;
+        }
+        activeDatasetSet.add(dataset);
+        if (!isDatasetActive(dataset)) {
```

Comment:
I got confused. What is the difference between "activeDatasetSet" and "isDatasetActive"?

Reply:
Seems isDatasetActive() checks datasetRefreshTimeMap. 
Can we just  check datasetRefreshTimeMap.containsKey() instead of isDatasetActive()?
Mixed the public method and private field makes it hard to read.